src/Microsoft.Extensions.ObjectPool/DefaultObjectPool.cs

```diff
@@ -23,6 +24,7 @@ public DefaultObjectPool(IPooledObjectPolicy<T> policy)
         public DefaultObjectPool(IPooledObjectPolicy<T> policy, int maximumRetained)
         {
             _policy = policy ?? throw new ArgumentNullException(nameof(policy));
+            _fastPolicy = policy as PooledObjectPolicy<T>;
```

Comment:
This is worth a comment, explaining that this base class was introduced in 2.1 to avoid the interface call

Reply:
Ah, now I added the comment above. Is the intention still clear or should I move the comment to this line?

Reply:
Looks good!